include/nuttx/fs/fs.h

```diff
@@ -227,6 +229,10 @@ struct file_operations
 
   CODE int     (*poll)(FAR struct file *filep, FAR struct pollfd *fds,
                        bool setup);
+  CODE ssize_t (*readv)(FAR struct file *filep, FAR const struct iovec *iov,
```

Comment:
could we move the new callback to the end? otherwise all drivers/fs has to been modified.

Reply:
where exactly do you mean by "the end"?

Reply:
@xiaoxiang781216 ping.
the current position is already meant to avoid driver/fs modifications in the common cases.
as mountpt_operations and file_operations share the entries, i don't think there is a better position.

Reply:
if we want keeping mountpt_operations and file_operations share the common fields, the new callbacks have to been added in the middle of struct, which means we must modify all fs and char drivers. So, I think that it may better to:
1.replace `read/write` with `readv/writev`
2.extract `buffer` and `buflen` from the first `iov` array for legacy drivers
3.improve the driver to support multiple `iov` step by step
BTW, we can add `pos` to `readv/writev` to fix the race condition in `readp/writep` too.

Reply:
> if we want keeping mountpt_operations and file_operations share the common fields, the new callbacks have to been added in the middle of struct, which means we must modify all fs and char drivers.

no.

* this PR doesn't need to modify all file_operations. eg. we don't need to modify g_pipe_fops.
* the number of file_operations in the tree is significantly larger than mountpt_operations.
  so it makes more sense to avoid modifications in the former than the latter.

> So, I think that it may better to: 1.replace `read/write` with `readv/writev` 2.extract `buffer` and `buflen` from the first `iov` array for legacy drivers 3.improve the driver to support multiple `iov` step by step BTW, we can add `pos` to `readv/writev` to fix the race condition in `readp/writep` too.

now you are suggesting the quite opposite, that is, modify more and more drivers and filesystems?
i'm confused.

Reply:
It's fine, if the change would avoid the modification to char driver.

Reply:
>  BTW, we can add pos to readv/writev to fix the race condition in readp/writep too.

i'm thinking to make readv/wirtev take something like bsd uio structure instead of taking iov/iovcnt directly
so that it can be extended later. how do you think?
```
struct uio {
  struct iovec *uio_iov;
  int iovcnt;
};
```
cf. https://man.freebsd.org/cgi/man.cgi?query=uio&sektion=9&format=html

Reply:
yes, looks good.

Reply:
> > BTW, we can add pos to readv/writev to fix the race condition in readp/writep too.
> 
> i'm thinking to make readv/wirtev take something like bsd uio structure instead of taking iov/iovcnt directly so that it can be extended later. how do you think?
> 
> ```
> struct uio {
>   struct iovec *uio_iov;
>   int iovcnt;
> };
> ```
> 
> cf. https://man.freebsd.org/cgi/man.cgi?query=uio&sektion=9&format=html

done